sendgrid.go

```diff
@@ -22,11 +23,17 @@ func GetRequest(key string, endpoint string, host string) rest.Request {
 	return request
 }
 
+// Client allows us to set a custom HTTP Client
+type Client struct {
+  rest.Client
+}
```

Comment:
I don't like the struct embedding here.

I'd advocate deprecating the API function in this package since it doesn't do anything.